### DISCUSSION TITLE
Sorting and Searching is not working

### DISCUSSION THREAD

**Discussion by Stevenking1981** [rappasoft/laravel-livewire-tables#934] [General]
Hello,

today I installed the package. Everything is working witout the searching and sorting.  Iam using the example with the UserTable.

`<?php

namespace App\Http\Livewire;

use App\Models\User;
use Illuminate\Database\Eloquent\Builder;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;

class UserTable extends DataTableComponent
{
    protected $model = User::class;

    public function configure(): void
    {
        $this->setPrimaryKey('id');
        $this->setPerPageAccepted([-1, 10, 25, 50, 100]);
    }

    public function columns(): array
    {
        return [
            Column::make('Name')
                ->sortable()
                ->searchable(),
            Column::make('E-mail', 'email')
                ->sortable()
                ->searchable(),
            Column::make('Verified', 'email_verified_at')
                ->sortable(),
        ];
    }

    public function query(): Builder
    {
        return User::query();
    }
}`

Also the pagination is not working.
I checked the installation guide twice that i didnt miss anything.

 Laravel Version ......... 9.33.0  
 PHP Version ............. 8.1.11

**Comment by dgillier** [comment]
Hi,

Maybe try:
`Column::make('Name',’name’)
            ->sortable()
            ->searchable(), `

Denis

**Comment by Stevenking1981** [comment]
Hello Denis,

i tried 

```
Column::make('Name','name')->sortable()->searchable(),

Column::make('E-mail', 'email')->sortable()->searchable(),

Column::make('Verified', 'email_verified_at')->sortable(),
```

but didnt change anything. I See all the data in the table but no "livewire"  features doesnt work.

greets
Steven

**Reply by dgillier** [reply]
Did you check in your browser console if you don't have a livewire error ?

**Reply by Stevenking1981** [reply]
Thanks @dgillier i had a typo :see_no_evil: I am sorry 

Greets
Steven

**Comment by AbhinavNmdo** [comment]
Try this snippet

`<?php

namespace App\Http\Livewire;

use App\Models\LoginMaster;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;
use App\Models\UserMaster;
use Exception;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\DB;
use Rappasoft\LaravelLivewireTables\Views\Filters\DateFilter;

class UserMasterTable extends DataTableComponent
{
    protected $model = UserMaster::class;

    public function configure(): void
    {
        $this->setPrimaryKey('id')
            ->setSingleSortingDisabled()
            ->setFilterLayoutSlideDown()
            ->setRememberColumnSelectionDisabled();

        $this->setDefaultSort('name', 'asc');

        $this->setBulkActions([
            'delete' => 'Delete',
        ]);
    }

    public function delete()
    {
        try {
            DB::beginTransaction();

            UserMaster::whereIn('id', $this->getSelected())->update([
                'is_active' => 0,
                'updated_by' => Auth::user()->user_master_id,
                'updated_at' => date('Y-m-d H:i:s')
            ]);

            LoginMaster::whereIn('user_master_id', $this->getSelected())->update([
                'is_active' => 0,
                'updated_by' => Auth::user()->user_master_id,
                'updated_at' => date('Y-m-d H:i:s')
            ]);

            $this->clearSelected();
            DB::commit();
            toastr()->success(__('auth.destroy', ['attr' => 'User']));
        } catch (Exception $th) {
            DB::rollBack();
            toastr()->error(__('auth.error'));
        }
    }

    public function builder(): Builder
    {
        return UserMaster::query()
                    ->where('is_active', 1);
    }

    public function columns(): array
    {
        return [
            Column::make('Name', 'name')
                ->sortable()
                ->collapseOnMobile()
                ->searchable(),
            Column::make('E-mail', 'email')
                ->sortable()
                ->collapseOnMobile()
                ->searchable(),
            Column::make('Mobile', 'mobile')
                ->sortable()
                ->searchable()
                ->collapseOnMobile(),
            Column::make('Joined At', 'created_at')
                ->sortable()
                ->collapseOnMobile()
                ->format(
                    fn ($value, $row, Column $column) => date_format(date_create($value), 'd M Y')
                ),
            Column::make('Action', 'id')
                ->format(function($value, $row, Column $column){
                    return view('admin.layout.action')->with([
                        'row' => $row
                    ]);
                }
                )->html()
        ];
    }

    public function filters(): array
    {
        return [
            DateFilter::make('Joined From')
                ->filter(function(Builder $builder, string $value) {
                    $builder->where('joined_at', '>=', $value);
                }),

            DateFilter::make('Joined To')
                ->filter(function(Builder $builder, string $value) {
                    $builder->where('joined_at', '<=', $value);
                }),
        ];
    }

    public function query(): Builder
    {
        return UserMaster::query();
    }
}
`
